diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5afa72e..de1ea97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -304,12 +304,10 @@ endif ()
 add_subdirectory (src/api)
 
 if (NLOPT_PYTHON)
-  find_package (PythonInterp)
-  find_package (PythonLibs ${PYTHON_VERSION_STRING} EXACT)
-  find_package (NumPy)
+  find_package (Python COMPONENTS Interpreter Development NumPy)
 endif ()
 
-if (NOT DEFINED INSTALL_PYTHON_DIR AND PYTHONINTERP_FOUND)
+if (NOT DEFINED INSTALL_PYTHON_DIR AND PYTHON_FOUND)
   execute_process ( COMMAND ${PYTHON_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(plat_specific=True, prefix='${CMAKE_INSTALL_PREFIX}'))"
                     OUTPUT_VARIABLE _ABS_PYTHON_MODULE_PATH
                     OUTPUT_STRIP_TRAILING_WHITESPACE )
