diff --git a/src/swig/CMakeLists.txt b/src/swig/CMakeLists.txt
index 78e2533..c4b9691 100644
--- a/src/swig/CMakeLists.txt
+++ b/src/swig/CMakeLists.txt
@@ -17,7 +17,7 @@ endif ()
 include_directories (${NLOPT_PRIVATE_INCLUDE_DIRS})
 set_source_files_properties (nlopt.i PROPERTIES CPLUSPLUS ON)
 
-if (NUMPY_FOUND AND PYTHONLIBS_FOUND)
+if (Python_FOUND)
 
   set (SWIG_MODULE_nlopt_python_EXTRA_DEPS nlopt-python.i numpy.i generate-cpp)
 
@@ -35,8 +35,8 @@ if (NUMPY_FOUND AND PYTHONLIBS_FOUND)
     set (nlopt_python "nlopt_python")
   endif ()
 
-  target_include_directories (${nlopt_python} PRIVATE ${PYTHON_INCLUDE_DIRS})
-  target_include_directories (${nlopt_python} PRIVATE ${NUMPY_INCLUDE_DIRS})
+  target_include_directories (${nlopt_python} PRIVATE ${Python_INCLUDE_DIRS})
+  target_include_directories (${nlopt_python} PRIVATE ${Python_NumPy_INCLUDE_DIRS})
 
   swig_link_libraries (nlopt_python ${nlopt_lib})
   target_link_libraries_with_dynamic_lookup (${nlopt_python} ${PYTHON_LIBRARIES})
